### ISSUE TITLE
weird kiosk message #2 

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#980]
Maybe related to #934 but it was a different message so I thought it warranted a new ticket. 

I clicked on the workstation to download it and got a weird kiosk message, so I opened the developer tools and got you the following screenshot:
![image](https://user-images.githubusercontent.com/67970402/110698063-f09b8400-81cb-11eb-81de-0834697c6df2.png)

I then exited the message and clicked on the workstation again, and was then able to download it normally.

**Comment by luizaogs** [comment]
And I forgot to tell you about the browser and operating system, of course. Still Chrome and macOS Big Sur 11.1.

**Comment by urapadmin** [comment]
- what you did not do, I suspect, is what the message asked you to: refresh the page? Or did you? 
 
- Was this clearly connected to you clicking on the workstation or is it possible that it popped up because you had this open for a while without doing anything and the clicking and the message were sort of coincidental? I know, difficult to discern. 
Why I ask is that this message is supposed to occur when your local session has timed out. But that would need idling.

**Comment by luizaogs** [comment]
Erm, no, I did not refresh the page. I'm not very good at following the directions in your messages, it seems. Not on purpose! Would that have led to different results?

I would be surprised if it occurred because of a session time out since it popped up right after I clicked on the workstation, which I did immediately after logging in.

**Comment by urapadmin** [comment]
> Would that have led to different results?
No. But it would have given you the feeling things are working according to a plan :o) In fact often enough with this internet crap it just helps to reload a page aka just do it again. 

> it popped up right after I clicked on the workstation, which I did immediately after logging in.
if you had just logged in, we can rule out the idling. 

You work on your following directions, I try to show messages in the right circumstances, which is clearly not the case here.

**Comment by luizaogs** [comment]
Ok well I got it again and this time it was a time out because I left it open while getting dinner, but then the message was also different and told me it was a time out, unlike the earlier one:

![image](https://user-images.githubusercontent.com/67970402/110714961-1aac7080-81e3-11eb-88f9-3105c21be1ba.png)

I did refresh this time (look at that!) and it led me back to the log in page.

**Comment by urapadmin** [comment]
Ja, that is a different message and it is really about the fact that the server refuses your timed-out session.

**Comment by urapadmin** [comment]
### @test kiosk 0.12
In connection with #1009 I have changed a few lines of code and tried to make messaging a bit more robust. While the second message you got is totally correct, the first message should not occur again easily. While you were testing this with the old workstation manager the issue could have arisen in any other module. You cannot test this in particular, just keep your eyes open if such "weird kiosk messages" come up again somewhere.

**Comment by luizaogs** [comment]
Ok well, just because I just told you I hadn't seen this in a while. A weirdly timed time out warning, since I'm currently downloading a workstation so I don't really see how I could have timed out? The download worked fine. Anyway, here are the developer tools.

![Screen Shot 2021-07-26 at 10 01 22 PM](https://user-images.githubusercontent.com/67970402/127078526-b7e89d59-1d8f-4c8f-9f2e-09643b0e0bf2.png)

**Comment by urapadmin** [comment]
That is a coincidence I cannot really address specifically. A download happens outside of my control (that's why I have to tell users that it 'might be already done' or so, because I cannot know) and so this can happen if nothing else goes on. I still have this open: #1157 and might look into it again. I am still not 100% sure what actions reset the timeouts. 
But I suppose we have to live with such an occassional message for v1. At least it is a real message and one can follow its instructions.

**Comment by luizaogs** [comment]
Can we close this? I haven't seen this in a long time.

**Comment by urapadmin** [comment]
definitely not a frequent issue
